pq/pq.go

```diff
@@ -71,53 +82,44 @@ func MakeRoot() [SzRoot]byte {
 // start transactions. An error is returned if the delegate is nil, the queue
 // header is invalid, some settings are invalid, or if some IO error occurred.
 func New(delegate Delegate, settings Settings) (*Queue, error) {
+	const op = "pq/new"
+
 	if delegate == nil {
-		return nil, errNODelegate
+		return nil, txerr.Op(op).Of(InvalidParam).Msg("delegate must not be nil")
 	}
 
-	accessor, err := makeAccess(delegate)
-	if err != nil {
-		return nil, err
+	accessor, k := makeAccess(delegate)
+	if k != NoError {
+		return nil, txerr.Op(op).Of(k).Err()
 	}
 
-	q := &Queue{accessor: accessor}
-
 	pageSize := delegate.PageSize()
-	pagePool := newPagePool(pageSize)
+
+	q := &Queue{
+		accessor: accessor,
+		settings: settings,
+		pagePool: newPagePool(pageSize),
+	}
+
+	// use pointer address as ID for correlating error messages
+	q.id = queueID(uintptr(unsafe.Pointer(q)))
```

Comment:
Can you share some more background here on why and how you correlate error messages?

Reply:
It's mostly for debugging purposes.

Each queue has it's own ID (in this case based on pointer). Transaction IDs are an uint64 increasing counter. E.g. in the context of transaction a many errors can occur. Each error generated within the same transaction will print the `txid`. Once a txid is known, one can search for all other related messages by txid.
In the future we might have multiple queues per spool file (e.g. due to versioning). Using the queue ID we can tell if one or multiple queues fail from time to time + one can correlate multiple errors to one queue. Would be nice if we can set the queue id from the outside and store queue ids in the spool file.